components/FilterList/index.js

```diff
@@ -0,0 +1,94 @@
+import { useEffect, useState } from "react";
+import ProjectsList from "../ProjectsList";
+import styled from "styled-components";
+
+export default function FilterList({ projects }) {
+  const complexities = [
+    { name: "All", color: "#070ff7" },
+    { name: "Beginner", color: "#3ecd5e" },
+    { name: "Intermediate", color: "#f9b234" },
+    { name: "Advanced", color: "#e44002" },
+  ];
+
+  const [filteredProjects, setFilteredProjects] = useState(projects);
```

Comment:
Why do you use a useState for this? To reduce complexity I recommend the following approach:
```js
  const [activeFilter, setActiveFilter] = useState("All");

  const filteredProjects =
    activeFilter === "All"
      ? projects
      : projects.filter((project) => project.complexity === activeFilter);

  function filterByComplexity(complexity) {
    setActiveFilter(complexity);
```
You won't need the useEffect anymore. And the cleaner (simpler) the code the less bugs in the future. Try to use as less useStates as possible.

Reply:
thanks for the hint, I will adjust immediately